avionics/common/include/tasks/restart_camera.hpp

```diff
@@ -0,0 +1,36 @@
+#ifndef RESTART_CAMERA_HPP_EA5200B75ECD4CC897C73B0CD1E8BA99
+#define RESTART_CAMERA_HPP_EA5200B75ECD4CC897C73B0CD1E8BA99
+#include "cameras.h"
+#include "scheduler.hpp"
+
+// REVIEW(akoen) these values have not tested and may need to be rocket specific.
+constexpr uint32_t CAMERA_POWEROFF_DURATION = 2000;
+constexpr uint32_t CAMERA_RESTART_INTERVAL = 20000;
+
+class RestartCamera {
+private:
+    Camera &cam_;
+    static bool camera_on;
+
+public:
+    RestartCamera(Camera &cam) : cam_(cam) {};
+
+    static void toggle_power(void *self) {
+        reinterpret_cast<RestartCamera *>(self)->toggle_power();
+    }
+private:
+
+    void toggle_power() {
+        Scheduler::Task &this_task = Scheduler::get_task(static_cast<int>(TaskID::RestartCamera));
+        if (camera_on) {
+            cam_.stop_record();
+            camera_on = false;
+            this_task.period = Scheduler::Duration{CAMERA_POWEROFF_DURATION};
+        } else {
+            cam_.start_record();
+            camera_on = true;
+            this_task.period = Scheduler::Duration{CAMERA_RESTART_INTERVAL};
+        }
+    }
```

Comment:
If you want, you can put the implentation of this class into the .cpp file - that'll slightly reduce compilation time